[ADD] polpo_vapi_integration: initial commit

qappsuy opened the pull request:
This module enables seamless Vapi.ai AI voice agent integration in Odoo, with out-of-the-box support. Call your voice AI agent created in Vapi.ai with Odoo to manage customer interactions, create leads, and log calls directly from your Odoo instance.

qappsuy committed:
[ADD] polpo_vapi_integration: initial commit

qappsuy committed:
[FIX] polpo_vapi_integration: OCA compliance fixes

- Fix manifest author and website keys for OCA standards
- Remove deprecated <data> tag in res_config_settings_views.xml
- Split long Python lines to comply with flake8
- Add i18n support to ValidationError messages
- Autoformat with black, autoflake, isort, prettier

qappsuy committed:
[FIX] polpo_vapi_integration: OCA js compliance fixes

qappsuy committed:
[FIX] polpo_vapi_integration: OCA js compliance fixes

qappsuy committed:
[FIX] polpo_vapi_integration: OCA compliance fixes

qappsuy committed:
[FIX] polpo_vapi_integration: OCA compliance fixes

qappsuy committed:
[FIX] polpo_vapi_integration: OCA compliance fixes

qappsuy committed:
chore: format code, fix lint, sync with OCA pre-commit

qappsuy committed:
[FIX] polpo_vapi_integration: OCA compliance fixes

etobella commented:
@qappsuy squash your commits in a single one please.

etobella commented on polpo_vapi_integration/static/src/js/vapi_integration_widget.js:
```diff
@@ -0,0 +1,223 @@
+$(document).ready(function () {
```
This is not OWL. Do it with OWL please

etobella commented on polpo_vapi_integration/LICENSE:
```diff
@@ -0,0 +1,17 @@
+This file is part of Odoo Vapi Call Integration module.
```
This is not necessary

etobella commented on polpo_vapi_integration/README.md:
```diff
@@ -0,0 +1,69 @@
+# Voice Agent with AI for Odoo – Vapi Integration
+
```
Use the readme folder as other modules

etobella commented on polpo_vapi_integration/views/vapi_integration_vapi_sdk.xml:
```diff
@@ -0,0 +1,13 @@
+<odoo>
+    <template id="vapi_integration_vapi_sdk" inherit_id="web.layout">
+        <xpath expr="//head" position="inside">
+            <script
```
Don't import external files. The library should be inside the module.

There are several reasons for that. The most important is that you avoid extra connections. Also, it ensures that the module will keep working if the library changes.

jelenapoblet replied:
@arielbarreiros96 , can you review this please ?

etobella reviewed (changes requested):
You shouldn't use your company name in the module name. In this case, it should be something like vapi_integration. However, in this repository, we stated should be named like like ai_oca_.... so, it should be ai_oca_vapi

qappsuy committed:
[IMP] ai_oca_vapi_integration: apply all PR review suggestions

[IMP] ai_oca_vapi_integration: addressed all review comments
- Moved Vapi SDK to local assets
- Reorganized README into readme/ folder
- Handled optional fields safely (e.g. social_linkedin)

Im not being able to implement OWL instead of JS, any help would be appreciated!

etobella commented on ai_oca_vapi_integration/static/src/js/vapi_integration_widget.js:
```diff
@@ -0,0 +1,225 @@
+$(document).ready(function () {
```
Still no OWL

etobella commented on ai_oca_vapi_integration/static/src/js/vapi_sdk.js:
```diff
@@ -0,0 +1,156 @@
+(function(oe){typeof define=="function"&&define.amd?define(oe):oe()})(function(){"use strict";var Tl=Object.defineProperty;var Cl=(oe,ve,Ee)=>ve in oe?Tl(oe,ve,{enumerable:!0,configurable:!0,writable:!0,value:Ee}):oe[ve]=Ee;var V=(oe,ve,Ee)=>(Cl(oe,typeof ve!="symbol"?ve+"":ve,Ee),Ee);var oe=typeof globalThis<"u"?globalThis:typeof window<"u"?window:typeof global<"u"?global:typeof self<"u"?self:{} ...
```
Should be in static/libs/vapi

etobella commented on setup/polpo_vapi_integration/odoo/addons/polpo_vapi_integration:
```diff
@@ -0,0 +1 @@
+../../../../polpo_vapi_integration
```
This file shouldn't be here

qappsuy committed:
[IMP] ai_oca_vapi_integration: apply PR review suggestions

Im still not being able to implement OWL instead of JS, any help would be appreciated!

qappsuy committed:
[IMP] ai_oca_vapi_integration: apply PR review suggestions
